test/controllers/species_lists_controller_test.rb

```diff
@@ -725,43 +668,124 @@ def test_construct_species_list_with_member_fields
+    assert_project_checks(@proj1.id => :unchecked, @proj2.id => :no_field)
+  end
+
+  # ............................
+  #    Edit and Update
+  # ............................
 
-  def test_edit_species_list
+  def test_edit
     spl = species_lists(:first_species_list)
     params = { id: spl.id.to_s }
     assert_equal("rolf", spl.user.login)
-    requires_user(
-      :edit,
-      :show,
-      params
-    )
+
+    requires_user(:edit, spl.id.to_s, params)
+    assert_response(:success)
     assert_edit_species_list
-    assert_form_action(
-      action: :edit,
-      id: spl.id.to_s,
-      approved_where: "Burbank, California, USA"
+    assert_select("form", { action: species_list_path,
+                            method: "post" }) do
+      assert_select("form input", { type: "hidden",
+                                    name: "id",
+                                    value: spl.id.to_s })
+      assert_select("form input", { type: "hidden",
+                                    name: "species_list_approved_where",
+                                    value: "Burbank, California, USA" })
+    end
+
+    spl = species_lists(:unknown_species_list)
+    login(users(:zero_user).login)
+    # TODO: use following path instead of action once helper paths are available
+    # get edit_species_list_path(params: { id: spl.id })
+    get(:edit, params: { id: spl.id })
+    assert_response(:redirect,
+                    "List non-member should be redirected upon attempt to edit")
+
+    proj = projects(:bolete_project)
+    assert(spl.projects.include?(proj))           # owned by bolete project
+    assert_not_equal(dick.id, spl.user_id)        # dick id not list owner
+    assert_equal([dick.id],
+                 proj.user_group.users.map(&:id)) # He's just a project member
+    login("dick")
+    # TODO: use following path instead of action once helper paths are available
+    # get edit_species_list_path(params: { id: spl.id })
+    get(:edit, params: { id: spl.id })
+    assert_response(
+      :success, "Any project member should be able to edit a project's lists"
     )
   end
 
-  def test_update_species_list_nochange
+  def test_presence_of_links_to_modify_species_list
+    spl = species_lists(:unknown_species_list)
+    proj = projects(:bolete_project)
+    assert_equal(mary.id, spl.user_id)            # owned by mary
+    assert(spl.projects.include?(proj))           # owned by bolete project
+    assert_equal([dick.id],
+                 proj.user_group.users.map(&:id)) # dick is a project member
+
+    # Prove non-member does not get links to edit or destroy list
+    login(users(:zero_user).login)
+    # TODO: use following path instead of action once helper paths are available
+    # get species_list_path(params: { id: spl.id })
+    get(:show, params: { id: spl.id })
+    assert_select("a:match('href',?)", %r{#{spl.id}\/edit}, count: 0)
+    assert_select("a[href*=destroy]", count: 0)
```

Comment:
These assertions pass (as they assert the non-existence of links). 
The remaining assertions in this method fail.